### ISSUE TITLE
Implement task inputs and outputs

### ISSUE THREAD

**Issue by fdasoghe** [jvoegele/gradle-android-plugin#6]
I've noticed that all the tasks from this plugin get always executed, even when their outputs are already up to date. 

For example, if I build my apk with androidPackageDebug and then run 
    gradle androidInstall

I get this output:
    :androidProcessResources
    :compileJava UP-TO-DATE
    :processResources UP-TO-DATE
    :classes UP-TO-DATE
    :jar UP-TO-DATE
    :proguard
    :androidPackageDebug
    :androidInstall

I think it could be cool (and very proficient) to have all tasks passing fast throught the whole lifecycle.
In order to accomplish this, maybe one has to declare the task's inputs and outputs, as described in the Gradle user guide, chapter 14.8.

**Comment by marcusb** [comment]
Here is some code that I use in build.gradle. (I don't use ProGuard here.)

```
// factor out dx into a separate task 
task dex(dependsOn: jar) << {
    logger.info("Converting compiled files and external libraries into ${intermediateDexFile}...")
    ant.apply(executable: ant.dx, failonerror: true, parallel: true) {
        arg(value: "--dex")
        arg(value: "--output=${intermediateDexFile}")
        inputs.files.each {
            fileset file: it
        }
    }
}
dex {
    inputs.file proguard.outJar
    inputs.files configurations.runtime.files
    outputs.files intermediateDexFile
}
// make androidPackage* depend on the dex task
task androidPackageRelease(overwrite: true, dependsOn: dex) << {
    logger.info("Packaging resources")
    sdkTools.aaptexec.execute(command: 'package')
    // ... etc
}
```

**Comment by fdasoghe** [comment]
Nice patch! I think it should be implemented into the plugin.

More generally speaking, maybe it could improve the usability and configurability of the plugin to adhere to Gradle's standards. I mean:
- using inputs and outputs file as said above (for performance and for next point)
- having all android\* tasks using same artifact, so that the entire lifecycle is more predictable. I refer here to the issue #10 where the androidPackageRelease task fails because it uses a filename different from the one used by the general method ApkBuilderTask_r7.execute() (that is "out.debug.unaligned.package").
- using predefined Gradle artifact names (from the user guide, table 15.1: baseName-appendix-version-classifier.extension)

**Comment by fdasoghe** [comment]
Fixed in 0.9.5. See the README.txt for instructions.
